Ginger/GingerCore/ALM/Qtest/ExportToQtest.cs

```diff
@@ -0,0 +1,594 @@
+                        result = "No matching Test Set was found in QC/ALM.";
+                    }
+                                    
+                }
+                if (result == string.Empty)
+                {
+                    result = "Export performed successfully.";
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }              
+            }
+            catch (Exception ex)
+            {
+                result = "Unexpected error occurred- "+ex.Message;
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to export execution details to QC/ALM", ex);
+                //if (!silentMode)
+                //    Reporter.ToUser(eUserMsgKey.ErrorWhileExportingExecDetails, ex.Message);
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Export Activities Group details to QC, can be used for creating new matching QC Test Case or updating an existing one
+        /// </summary>
+        /// <param name="activitiesGroup">Activities Group to Export</param>
+        /// <param name="mappedTest">The QC Test Case which mapped to the Activities Group (in case exist) and needs to be updated</param>
+        /// <param name="uploadPath">Upload path in QC Test Plan</param>
+        /// <param name="result">Export error result</param>
+        /// <returns></returns>
+        public static bool ExportActivitiesGroupToQC(ActivitiesGroup activitiesGroup, Test mappedTest, string uploadPath, ObservableList<ExternalItemFieldBase> testCaseFields, ref string result)
+        {
+            Test test;
+
+            try
+            {
+                if (mappedTest == null)
+                {
+                    //##create new Test Case in QC
+                    TestFactory TestF = (TestFactory)mTDConn.TestFactory;
+                    test = (Test)TestF.AddItem(System.DBNull.Value);
+                    test.Type = "MANUAL";
+
+                    //set the upload path
+                    TreeManager treeM = (TreeManager)mTDConn.TreeManager;
+                    ISysTreeNode testParentFolder = (ISysTreeNode)treeM.get_NodeByPath(uploadPath);
+                    test["TS_SUBJECT"] = testParentFolder.NodeID;
+                    
+                }
+                else
+                {
+                    //##update existing test case
+                    test = ImportFromQC.GetQCTest(activitiesGroup.ExternalID);
+
+                    //delete the un-needed steps
+                    DesignStepFactory stepF = test.DesignStepFactory;
+                    List stepsList = stepF.NewList("");
+                    foreach (DesignStep step in stepsList)
+                    {
+                        if (activitiesGroup.ActivitiesIdentifiers.Where(x => x.IdentifiedActivity.ExternalID == step.ID.ToString()).FirstOrDefault() == null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)